runtime/ast/unmarshal.go

```diff
@@ -0,0 +1,901 @@
+package ast
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/onflow/cadence/runtime/common"
+)
+
+func jsonMarshalAndVerify(i interface{}) (b []byte, err error) {
+	if b, err = json.Marshal(i); err == nil {
+		expected := string(b)
+		roundtripped := reflect.New(reflect.TypeOf(i)).Interface()
+		if err = json.Unmarshal(b, &roundtripped); err == nil {
+			if b, err = json.Marshal(roundtripped); err == nil {
+				actual := string(b)
+				if expected != actual {
+					return nil, fmt.Errorf("un/marshal failed:\n%s\n%s\n", expected, actual)
+				}
+			}
+		}
+	}
+	return
+}
+
+//////////////////////////////////////////////////////////////////////////////
+
+var accessKindMap = map[string]Access{
```

Comment:
Just like with `declarationKindMap`, consider moving these maps here next to their respective types. That way, we won't forget updating these maps when the enums change.